src/utils.py

```diff
@@ -666,26 +666,27 @@ def extract_preds_and_types(nsrts: Collection[NSRT]) -> Tuple[
     return preds, types
 
 
-def filter_static_nsrts(ground_nsrts: Collection[_GroundNSRT],
-                        atoms: Collection[GroundAtom]) -> List[
-                            _GroundNSRT]:
-    """Filter out ground NSRTs that don't satisfy static facts.
+_O = TypeVar('_O', bound=Union[_GroundNSRT, _GroundSTRIPSOperator])
```

Comment:
@ronuchit I think we can probably refactor a number of things using this pattern, yeah?

Reply:
this pattern doesn't work as intended bc `bound` isn't what we actually want, but see #220